### ISSUE TITLE
cannot install 3.12 on 14.1.1 new mbp - Modules/_blake2.cpython-312-darwin.so: No such file or directory

### ISSUE THREAD

**Issue by robbiemu** [pyenv/pyenv#2850]
```
robbie@macbook-pro ~ % env PYTHON_CONFIGURE_OPTS="--enable-framework --enable-optimizations --with-lto" PYTHON_CFLAGS='-march=native -mtune=native' pyenv install 3.12.0
python-build: use openssl@3 from homebrew
python-build: use readline from homebrew
Downloading Python-3.12.0.tar.xz...
-> https://www.python.org/ftp/python/3.12.0/Python-3.12.0.tar.xz
Installing Python-3.12.0...
python-build: use readline from homebrew
python-build: use zlib from xcode sdk

BUILD FAILED (OS X 14.1.1 using python-build 20180424)

Inspect or clean up the working tree at /var/folders/9h/_r86cs8177x6hmpxy0fg473h0000gp/T/python-build.20231124165832.17971
Results logged to /var/folders/9h/_r86cs8177x6hmpxy0fg473h0000gp/T/python-build.20231124165832.17971.log

Last 10 log lines:
/usr/bin/install -c -m 755 Modules/zlib.cpython-312-darwin.so /Users/robbie/.pyenv/versions/3.12.0/lib/python3.12/lib-dynload/zlib.cpython-312-darwin.so
/usr/bin/install -c -m 755 Modules/_dbm.cpython-312-darwin.so /Users/robbie/.pyenv/versions/3.12.0/lib/python3.12/lib-dynload/_dbm.cpython-312-darwin.so
/usr/bin/install -c -m 755 Modules/readline.cpython-312-darwin.so /Users/robbie/.pyenv/versions/3.12.0/lib/python3.12/lib-dynload/readline.cpython-312-darwin.so
/usr/bin/install -c -m 755 Modules/_md5.cpython-312-darwin.so /Users/robbie/.pyenv/versions/3.12.0/lib/python3.12/lib-dynload/_md5.cpython-312-darwin.so
/usr/bin/install -c -m 755 Modules/_sha1.cpython-312-darwin.so /Users/robbie/.pyenv/versions/3.12.0/lib/python3.12/lib-dynload/_sha1.cpython-312-darwin.so
/usr/bin/install -c -m 755 Modules/_sha2.cpython-312-darwin.so /Users/robbie/.pyenv/versions/3.12.0/lib/python3.12/lib-dynload/_sha2.cpython-312-darwin.so
/usr/bin/install -c -m 755 Modules/_sha3.cpython-312-darwin.so /Users/robbie/.pyenv/versions/3.12.0/lib/python3.12/lib-dynload/_sha3.cpython-312-darwin.so
/usr/bin/install -c -m 755 Modules/_blake2.cpython-312-darwin.so /Users/robbie/.pyenv/versions/3.12.0/lib/python3.12/lib-dynload/_blake2.cpython-312-darwin.so
install: Modules/_blake2.cpython-312-darwin.so: No such file or directory
make: *** [sharedinstall] Error 71
```

PYENV_DEBUG trace https://gist.github.com/robbiemu/c93d9c0c12ca03519387fb5be6fb16fd
I do see blake2 module getting built, with no error in the log:
```
clang -bundle -undefined dynamic_lookup -L/opt/homebrew/opt/readline/lib -L/opt/homebrew/opt/readline/lib -L/Users/robbie/.pyenv/versions/3.12.0/lib -L/opt/homebrew/lib -L/usr/local/lib  -L/opt/homebrew/lib -Wl,-rpath,/opt/homebrew/lib -L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib -L/opt/homebrew/opt/readline/lib -L/opt/homebrew/opt/readline/lib -L/Users/robbie/.pyenv/versions/3.12.0/lib -L/opt/homebrew/lib -L/usr/local/lib  -L/opt/homebrew/lib -Wl,-rpath,/opt/homebrew/lib -L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib -flto=thin -Wl,-export_dynamic -Wl,-object_path_lto,"Modules/_blake2.cpython-312-darwin.so".lto -g -fprofile-instr-generate  Modules/_blake2/blake2module.o Modules/_blake2/blake2b_impl.o Modules/_blake2/blake2s_impl.o -L/usr/local/Cellar/libb2/0.98.1/lib -lb2  -o Modules/_blake2.cpython-312-darwin.so
rm -f Modules/expat/libexpat.a
...
```
but then it is apparently removed:
```
DYLD_FRAMEWORK_PATH=/var/folders/9h/_r86cs8177x6hmpxy0fg473h0000gp/T/python-build.20231124170639.39247/Python-3.12.0 ./python.exe -E -c 'import sys ; from sysconfig import get_platform ; print("%s-%d.%d" % (get_platform(), *sys.version_info[:2]))' >platform
[ERROR] _blake2 failed to import: dlopen(/private/var/folders/9h/_r86cs8177x6hmpxy0fg473h0000gp/T/python-build.20231124170639.39247/Python-3.12.0/build/lib.macosx-14.1-arm64-3.12/_blake2.cpython-312-darwin.so, 0x0002): symbol not found in flat namespace '_blake2b_final'
The necessary bits to build these optional modules were not found:
_gdbm                 _tkinter                                 
To find the necessary bits, look in configure.ac and config.log.

Following modules built successfully but were removed because they could not be imported:
_blake2                                                        
```

**Comment by robbiemu** [comment]
actually I think this issue is related, and may point to the root problem:

https://github.com/pyenv/pyenv/issues/2573

I did install x64 arch brew for gaming, to /usr/local, and then arm64 arch brew to /opt

**Comment by native-api** [comment]
https://github.com/pyenv/pyenv/wiki/Common-build-problems#on-apple-silicon-when-building-for-arm64-a-dependency-is-present-in-x64-homebrew-but-not-arm64-homebrew

**Comment by leventyalcin** [comment]
I’m late to the party, but I encountered the same issue here and in #2805. None of the solutions I found online worked.

After trying many approaches, I attempted to run pyenv without activating my bash profile, which allowed me to narrow down potential causes one by one. I’m still unsure of the root cause, but having `"$(brew --prefix binutils)/bin"` in PATH was triggering the issue. Once I removed the `binutils` bin directory from PATH, I was able to install any Python version successfully.

I hope this helps someone else!
